lib/WUI/http/httpd_post.c

```diff
@@ -229,27 +229,34 @@ err_t httpd_post_begin(void *connection,
     uint32_t api_key_tag_length = strlen(API_KEY_TAG);
     char *api_key_start = lwip_strnstr(http_request, API_KEY_TAG, http_request_len);
 
+    if (upload.phase != UPLOAD_PROCESS_NONE) {
+        uri = "/409";
+        goto invalid;
+    }
+
     if (api_key_start == NULL) {
-        uri = "/401\0";
+        uri = "/401";
         goto invalid;
     } else {
         const char *api_key = wui_get_api_key();
         uint32_t token_length = strlen(api_key);
-        if (memcmp(api_key, api_key_start + api_key_tag_length, token_length) != 0) {
-            uri = "/401\0";
+
+        if (lwip_strnicmp(api_key, api_key_start + api_key_tag_length, token_length) != 0
+            || (api_key_start[api_key_tag_length + token_length] != '\r' && api_key_start[api_key_tag_length + token_length + 1] != '\n')) {
```

Comment:
Is the newline in-range for sure? Could our buffer end up with the end of the token?

Reply:
I have a safer fix (I hope) in the branch where I'm fixing UBs, so let's not dwell on these. The question if we want to authenticate case-sensitive or insensivite still stands, though.

Reply:
perfect